server/routes/reports/createReport.ts

```diff
@@ -0,0 +1,226 @@
+      const now = new Date()
+      if (incidentDateAndTime > now) {
+        const error = new BaseController.Error('incidentDate', {
+          key: 'incidentDate',
+          message: 'Enter a date and time in the past',
+        })
+        next({ incidentDate: error })
+        return
+      }
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    } catch (e) {
+      /* empty */
+    }
+
+    super.validate(req, res, next)
+  }
+
+  /**
+   * Combine date and time strings into a Date
+   * @throws {Error} when either date or time cannot be parsed
+   */
+  private buildIncidentDateAndTime(incidentDate: string, incidentTime: string): Date {
+    const incidentDateAndTime = parseDateInput(incidentDate)
+    const time = parseTimeInput(incidentTime)
+    incidentDateAndTime.setHours(time.hours)
+    incidentDateAndTime.setMinutes(time.minutes)
+    return incidentDateAndTime
+  }
+
+  protected errorMessage(error: FormWizard.Error): string {
+    if (error.key === 'incidentDate') {
+      return 'Enter a date'
+    }
+    if (error.key === 'incidentTime') {
+      return 'Enter a time'
+    }
+    if (error.key === 'description') {
+      return 'Enter a description'
+    }
+    return super.errorMessage(error)
+  }
+
+  async successHandler(
+    req: FormWizard.Request<CreateReport, Step2>,
+    res: Express.Response,
+    next: Express.NextFunction,
+  ): Promise<void> {
+    const allValues = this.getAllValues(req)
+
+    const { type, description, incidentDate, incidentTime } = allValues
+    const incidentDateAndTime = this.buildIncidentDateAndTime(incidentDate, incidentTime)
+    const typeDetails = getTypeDetails(type)
+    const title = `Report: ${typeDetails.description.toLowerCase()}`
+
+    try {
+      const report = await res.locals.apis.incidentReportingApi.createReport({
+        type,
+        incidentDateAndTime,
+        title,
+        description,
+        prisonId: res.locals.user.activeCaseLoad.caseLoadId,
+        createNewEvent: true,
+      })
+      logger.info(`Report ${report.reportReference} created`)
+      res.locals.createdReport = report
+
+      // TODO: complete form, forget values
+      this.setStepComplete(req, res)
+      // req.journeyModel.reset()
+      // req.sessionModel.reset()
+      // this.resetJourneyHistory(req, res)
+
+      super.successHandler(req, res, next)
+    } catch (e) {
+      logger.error({ err: e }, 'Report could not be created: %j', e)
+      const err = this.convertIntoValidationError(e)
+      // TODO: find a different way to report whole-form errors rather than attaching to specific field
+      this.errorHandler({ incidentDate: err }, req, res, next)
+    }
+  }
+
+  getNextStep(req: FormWizard.Request<CreateReport, Step2>, res: Express.Response): string | undefined {
+    // if a report was successfully created, redirect to it
+    if ('createdReport' in res.locals && res.locals.createdReport) {
```

Comment:
Equivalent to this no?

```JavaScript
if (res.locals?.createdReport) {
  // ;)
}
```

Reply:
except this makes eslint happy

Reply:
I, for one, _**NOT**_ welcome our new robot overlords...